cli/src/pcluster/resources/imagebuilder/parallelcluster_validate.yaml

```diff
@@ -0,0 +1,507 @@
+                if [ "${cuda_output}" != "release ${cuda_ver}" ]; then
+                  echo "CUDA installed incorrectly! Installed ${cuda_output} but expected ${cuda_ver}"
+                  exit 1
+                else
+                  echo "CUDA nvcc test passed, $cuda_output"
+                fi
+
+                # Test deviceQuery
+                echo "Testing CUDA install with deviceQuery..."
+                /usr/local/cuda-${cuda_ver}/extras/demo_suite/deviceQuery | grep -o "Result = PASS"
+                [[ $? -ne 0 ]] && echo "CUDA deviceQuery test failed" && exit 1
+                echo "CUDA deviceQuery test passed"
+                echo "Correctly installed CUDA ${cuda_output}"
+              fi
+
+      # FabricManager
+      - name: FabricManager
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -vx
+              if [ {{ validate.NvSwitches.outputs.stdout }} -gt 1 ]; then
+                echo "test fabric-manager daemon"
+                systemctl show -p SubState nvidia-fabricmanager | grep -i running
+                [[ $? -ne 0 ]] && echo "fabric-manager daemon test failed" && exit 1
+                echo "NVIDIA Fabric Manager service correctly started"
+              fi
+
+      # CloudWatch
+      - name: CloudWatch
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -vx
+              /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a status | grep status | grep stopped
+              [[ $? -ne 0 ]] && echo "amazon-cloudwatch-agent is not stopped" && exit 1
+
+      # FSx Lustre
+      - name: FSxLustre
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -vx
+              OS='{{ validate.OperatingSystemName.outputs.stdout }}'
+
+              if [ {{ validate.LustreSupported.outputs.stdout }} == true ]; then
+                if [ ${OS} == centos7 ]; then
+                  echo "check for lustre libraries"
+                  rpm -qa | grep lustre-client
+                elif [ ${OS} == ubuntu1804 ]; then
+                  echo "check for lustre libraries"
+                  dpkg -l | grep lustre
+                fi
+              fi
+              [[ $? -ne 0 ]] && echo "check lustre libraries failed" && exit 1
+
+      # PIP
+      - name: PIP
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -vx
+              COOKBOOK_PIP_VERSION=$({{ validate.CookbookVirtualenvPath.outputs.stdout }}/bin/pip show pip 2> /dev/null | grep -i "^Version:" | tr -s ' ' | cut -d' ' -f2) || echo "NOT_INSTALLED"
+              NODE_PIP_VERSION=$({{ validate.NodeVirtualenvPath.outputs.stdout }}/bin/pip show pip 2> /dev/null | grep -i "^Version:" | tr -s ' ' | cut -d' ' -f2) || echo "NOT_INSTALLED"
+
+              compare_version(){
+                CURRENT_VERSION="$1"
+                PEQUIRED_VERSION="19.3"
```

Comment:
type in var name?

Reply:
we can define a constant also here

Reply:
the component size is not enough to define a new constant